crypto-common/src/lib.rs

```diff
@@ -164,7 +164,7 @@ pub trait KeyInit: KeySizeUser + Sized {
     #[inline]
     fn generate_key(mut rng: impl CryptoRng + RngCore) -> Key<Self> {
         let mut key = Key::<Self>::default();
-        rng.fill_bytes(&mut key);
+        rng.fill_bytes(key.as_mut());
```

Comment:
It seems `GenericArray` has `Deref`/`DerefMut` impls with a target of `[T]`, whereas `hybrid-array` has a target of `ArraySize::ArrayType`, which is what allowed this to work before without the explicit `as_mut()`.

I think there might be some wisdom to that, and it would simplify updating existing code. Access to the inner array is still available via `AsRef`/`AsMut`, `Borrow`/`BorrowMut`, or `.0` since the inner array field is `pub`.

Reply:
Updated `hybrid-array` to `Deref` to a slice target